code/modules/mob/living/living.dm

```diff
@@ -8,6 +8,8 @@
 		set_name()
 	var/datum/atom_hud/data/human/medical/advanced/medhud = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
 	medhud.add_atom_to_hud(src)
+	var/datum/atom_hud/data/human/blood/blood_hud = GLOB.huds[DATA_HUD_BLOOD]
+	blood_hud.add_atom_to_hud(src)
```

Comment:
Likewise we don't need to add every living mob to the hud, only ones with blood
```suggestion
	if(blood_volume)
		var/datum/atom_hud/data/human/blood/blood_hud = GLOB.huds[DATA_HUD_BLOOD]
		blood_hud.add_atom_to_hud(src)
```

Reply:
I'll probably also check get_bloodtype()? Maybe not since it's never null I think? I also need to make sure that carbons are added regardless of if they have blood at init. I also need to implement checks for TRAIT_NOBLOOD. This is gonna take a while, isn't it...

Reply:
Okay, I've added a simple check for initial(blood_volume) here, just like what I did in Life().